manager/controllers/baseline_update.go

```diff
@@ -0,0 +1,152 @@
+}
+
+type BaselineConfig struct {
+	ToTime string `json:"to_time"`
+}
+
+type AccountID struct {
+	RhAccountID int
+}
+type UpdateBaselineResponse struct {
+	BaselineID int
+}
+
+// @Summary Update a baseline for my set of systems
+// @Description Update a baseline for my set of systems
+// @ID updateBaseline
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    baseline_id path    int                  true "Baseline ID"
+// @Param    body        body    UpdateBaselineRequst true "Request body"
+// @Success 200 {object} UpdateBaselineResponse
+// @Router /api/patch/v1/baselines/{baseline_id} [post]
+func BaselineUpdateHandler(c *gin.Context) {
+	account := AccountID{
+		RhAccountID: c.GetInt(middlewares.KeyAccount),
+	}
+
+	var req UpdateBaselineRequst
+	if err := c.ShouldBindJSON(&req); err != nil {
+		LogAndRespBadRequest(c, err, "Invalid request body")
+	}
+
+	baselineID := c.Param("baseline_id")
+	if baselineID == "" {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "baseline_id param not found"})
+		return
+	}
+
+	var exists int64
+	err := database.Db.Model(&models.Baseline{}).
+		Where("id = ? ", baselineID).Count(&exists).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+	}
+	if exists == 0 {
+		LogAndRespNotFound(c, errors.New("Baseline not found"), "Baseline not found")
+		return
+	}
+
+	config, err := json.Marshal(&req.Config)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid config")
+	}
+
+	ID, err := strconv.Atoi(baselineID)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid baseline id")
+	}
+
+	baseline := models.Baseline{
+		ID:          ID,
+		Name:        req.Name,
+		Config:      config,
+		RhAccountID: account.RhAccountID,
+	}
+
+	err = buildUpdateBaselineQuery(baseline, req.InventoryIDs, account, c)
+	if err != nil {
+		LogAndRespError(c, err, "Database error")
+	}
+
+	c.JSON(http.StatusOK, baseline.ID)
+}
+
+type SystemPlatform struct {
+	*models.SystemPlatform
+}
+
+func sortAssociations(inventoryIDs map[string]string) ([]string, []string) {
+	var newAssociations []string
+	var obsoleteAssociations []string
+	for key, value := range inventoryIDs {
+		if value == "true" {
+			newAssociations = append(newAssociations, key)
+		} else {
+			obsoleteAssociations = append(obsoleteAssociations, key)
+		}
+	}
+
+	return newAssociations, obsoleteAssociations
+}
+
+func buildUpdateBaselineQuery(baseline models.Baseline, inventoryIDs map[string]string,
+	account AccountID, c *gin.Context) error {
+	data := map[string]interface{}{}
+	data["name"] = baseline.Name
+	data["config"] = baseline.Config
+	newAssociations, obsoleteAssociations := sortAssociations(inventoryIDs)
+
+	tx := database.Db.WithContext(base.Context).Begin()
+	defer tx.Rollback()
+
+	if err := tx.Model(models.Baseline{}).Where("id = (?)", baseline.ID).Updates(&data).Error; err != nil {
+		LogAndRespError(c, err, "Database error")
+		return err
+	}
+
+	associationsQuery := tx.Model(models.SystemPlatform{}).
```

Comment:
This is quite long expression used twice, so create a method for that please:
~~~golang
func updateSystemsBaselineID(rhAccountID int, inventoryIDs []string, baselineID interface) error 
~~~